Add alpine-consul-apache

matthewvalimaki opened the pull request:
Copy of alpine-consul-nginx and does what the name implies.
```
"script": "wget -s http://localhost:80 > /dev/null 2>&1 || exit 2"
```
hits default index.html that is provided with the apache2 apk. I opted not to add a vhost for `/ping` as this was readily available and adding `/ping` is not as simple as with nginx.

smebberson commented on alpine-apache/Dockerfile:
```diff
@@ -0,0 +1,14 @@
+FROM smebberson/alpine-base:1.1.0
```
This has since been upgraded to `alpine-base:1.2.0`. Would you mind updating accordingly?

smebberson commented on alpine-apache/Dockerfile:
```diff
@@ -0,0 +1,14 @@
+# Install apache
+RUN apk add --update apache2 apache2-utils && \
+    rm -rf /var/cache/apk/* && \
+    ln -sf /dev/stdout /var/log/apache2/access.log && \
```
I plan on upgrading `alpine-consul-nginx` to not automatically stream logs to `stdout` and `stderr` as per `alpine-nginx`. It's an easy two lines in a downstream `Dockerfile` to stream them if you wish. What do you think @matthewvalimaki, should the `alpine-apache` image work the same?

smebberson commented on alpine-apache/README.md:
```diff
@@ -0,0 +1,93 @@
+
+Versions
+--------
+
```
Could you please add information about `v1.0.0` here?

matthewvalimaki replied:
Added. But I do think you need to fix it after accepting PR as the 1.0.0 link points at something right now that doesn't exist.

smebberson commented on alpine-apache/root/etc/services.d/apache/finish:
```diff
@@ -0,0 +1,8 @@
+
+# only tell s6 to bring down the entire container, if it isn't already doing so
+# http://skarnet.org/software/s6/s6-supervise.html
+if -n { s6-test $# -ne 0 }
```
This code has since been changed. You can see the updated code in https://github.com/smebberson/docker-alpine/blob/master/alpine-nginx/root/etc/services.d/nginx/finish. Would you mind updating accordingly? I am going to be bringing those changes to `alpine-consul-nginx` (etc).

smebberson commented on alpine-apache/tag:
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env sh
```
This file is no longer required. Managing so many `tag` scripts was becoming cumbersome. I've centralised this to one `tag` script at the root of the repository which now accepts the image name.

smebberson commented:
@matthewvalimaki, this is a great addition also! I've made some notes. Are you happy to go through and make the recommended changes?

matthewvalimaki commented:
All notes made sense and were incorporated. Let me know if you want me to clean it up.

smebberson commented:
@matthewvalimaki, I just pulled this down and ran everything. It's working great. Just wanted to consider something and get your thoughts.

With the nginx version of this, I created the route `/ping` and the nginx config, I've silenced that url so that it doesn't pollute the logs (or stdout).

This one is doing just that. Every time the `consul` health check takes place it's dumping the following to stdout:

```
127.0.0.1 - - [30/Nov/2015:02:06:20 +0000] "GET / HTTP/1.1" 200 263 "-" "Wget"
```

Do you think it's worth going to the trouble of creating a `ping.local` virtual host something, and silencing those logs? Then the `consul` health check would use that domain.

In real life, how are you extending this image?

matthewvalimaki commented:
@smebberson I've updated the setup a bit: I removed logging to `/dev/stdout` per your comments. I've also added `env=!dontlog` and `/etc/apache2/conf.d/ping.conf` to hide `GET /` from logs. No vhost used but does modify the default CustomLog (access log) behavior by adding directive `!dontlog`.

Let me know what you think of this instead of vhost.

matthewvalimaki commented:
@smebberson forgot to answer your question
> In real life, how are you extending this image?

For now I am simply extending it with PHP and mounting volumes from data only containers.

smebberson commented:
@matthewvalimaki. This has all been merged. Images are now available on Docker Hub. Thanks for this contribution! I changed a few things, and updated the documentation. Take a look at the last few commits to see what I changed (I hope you don't mind the change to alpine-consul-apache health checks, the configuration the way it was, was stopping all Apache access logs, not just those that hit the ping route).

Welcome aboard as the second biggest contributor to the project! https://github.com/smebberson/docker-alpine/graphs/contributors

matthewvalimaki commented:
@smebberson I don't mind at all! What matters is getting the image up. The changes are good by the way. Oh duh, yeah. The configuration for dontlog was horrible, now when I stop to think about it.

Thank you for these images :)
